client/src/main/java/io/pravega/client/stream/impl/ReaderGroupState.java

```diff
@@ -90,6 +90,19 @@
     @GuardedBy("$lock")
     private final Map<SegmentWithRange, Long> lastReadPosition;
     private final Map<Segment, Long> endSegments;
+    @Getter
+    private final long generation;
+    @Getter
+    private final ConfigState configState;
+    @Getter
+    private final ReaderGroupConfig newConfig;
+
+    public enum ConfigState {
+        INITIALIZING,
+        DELETING,
```

Comment:
one suggestion - we should add "DELETED" state after we complete the work for deleting. 

and we can add the logic that the RG stream can be deleted only if its ConfigState is deleted. this ensures that RG stream is not deleted prematurely without completing the deletion workflow.